docs/sources/operations/automatic-stream-sharding.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Automatic Stream Sharding
+menuTitle: Automatic stream sharding
+description: Automatic stream sharding can control issues around the per-stream rate limit
+weight: 110
+---
+
+# Automatic stream sharding
+
+Automatic stream sharding will attempt to keep streams under a `desired_rate` by adding new labels and values to
+existing streams. When properly tuned, this should eliminate issues where log producers are rate limited due to the
```

Comment:
nit: This clearly paint the picture from the customer point of view (avoiding per stream rate limit). Curious should we also make it visible how it helps Loki operators? By mentioning (or something similar) like

"Without automatic stream sharding, any high volume streams can easily overload ingestion path of Loki thus may end up in OOMing of particular ingesters (that owns the stream). With automatic stream sharding, this high volume streams are sharded further by spreading load across other ingesters without overloading any specific ingesters".

 May be wording are poor, just trying to communicate the gist :)

Reply:
Added a section like this lmkwyt